packages/siwe/lib/client.ts

```diff
@@ -426,21 +426,10 @@ export class SiweMessage {
     }
 
     /** EIP-55 `address` check. */
-    if (!isEIP55Address(this.address)) {
-      throw new SiweError(
-        SiweErrorType.INVALID_ADDRESS,
-        getAddress(this.address),
-        this.address
-      );
-    }
+    // this test is done in siwe-parser, callback function "address"
```

Comment:
If all the checks done in this function are done as part of the parsing, then I suggest no to modify this function but instead move its call in the constructor to only the case where it's initialised from an object.

Reply:
If I understand this correctly you still want to validate the EIP-55 address and the URI in the constructor block for the message object. The EIP-55 address will be simple enough, but I'll need to build a special validator that validates only the URI and not an entire message. The ABNF grammar object, `siwe-grammar.ts` and the existing call back functions give me everything I need for that, but it will require having an option in the `ParsedMessage` constructor to parse only the URI. How about the resource URIs? Will they need to be validated as well?

Reply:
I need to write a new `isUri()` function to use in the object block of the constructor. I just need to assemble it from existing pieces which won't be difficult. And, of course, do the unit testing. I'll need a few more days on this.

Reply:
The latest commit should resolve this conversation. An `isUri()` function has been added, unit tested and the message block of the `SiweMessage` constructor updated to test the EIP-55 address, the URI and all of the resource URIs if any are present. However, see the additional question in the following conversation.

Reply:
You've extracted the URI and EIP55 validation out of the validation function, but remains:
- domain
- version
- nonce
- dates

These can (or already are) validated in the parser, so can't the validate function be left untouched and only called when constructing the message from an object?

Reply:
Sorry, I missed the proper flow of the validation in the message object case. In particular I missed the `validateMessage()` function. All I really needed to do was replace the valid-url isUri() function with the new APG isUri() function there. All other validations will then remain the same. I can easily make that change. However, I notice there is no validation of the resources URIs. Do you have any need for that? It is easily added.

Here’s a better suggestion. Remove the `validateMessage()` function altogether and replace it with
> // this.validateMessage();
> new ParsedMessage(this.prepareMessage());

I’ve tried it and it passes all of my objects unit tests. This will also validate all of the resources URIs. (NOTE: that perpareMessage() needs a fix for the “empty statement” case that #30 pointed out was allowed in the spec.)

Reply:
BTW all of the `client.test.ts` unit tests pass with this fix also. But I have a question. When I run jest on `client.test.ts` it prints out several screens of red error text and a lot of warnings that about `validate() has been deprecated,` But then at the end it shows that all tests have passed. What's a person to make of this? Does something need to be fixed here?

Reply:
> it prints out several screens of red error text and a lot of warnings

This PR will require a new major version and the deprecated function will be removed then, so that will remove those warnings.

> Here’s a better suggestion. Remove the validateMessage() function altogether and replace it with `new ParsedMessage(this.prepareMessage());`

I was worried we would lose the information provided by `SiweErrorType` (e.g. if a nonce isn't long enough it would clearly state so), and we do a bit but it still points to the field (e.g. now it says something like "line 9: invalid nonce") so I'm ok with this change. But could you remove the variants from `SiweErrorType` that aren't used anymore?

Reply:
Related hot take, EIP55 validation should be optional https://github.com/spruceid/siwe/issues/53